icicle/CMakeLists.txt

```diff
@@ -44,20 +44,23 @@ if (NOT BUILD_TESTS)
     icicle
     appUtils/msm/msm.cu
     appUtils/ntt/ntt.cu
+    appUtils/lde/lde.cu
   )
   target_compile_options(icicle PRIVATE -c)
 
   add_custom_command(
     TARGET icicle
     POST_BUILD
-    COMMAND ${CMAKE_OBJCOPY} ARGS --redefine-sym msm_cuda=${CURVE}_msm_cuda --redefine-sym msm_internal_cuda=${CURVE}_msm_internal_cuda ${PROJECT_BINARY_DIR}/CMakeFiles/icicle.dir/appUtils/msm/msm.cu.o
-  )
-  add_custom_command(
-    TARGET icicle
-    POST_BUILD
-    COMMAND ${CMAKE_AR} ARGS -rcs ${LIBRARY_OUTPUT_DIRECTORY}/libingo_${CURVE}.a ${PROJECT_BINARY_DIR}/CMakeFiles/icicle.dir/appUtils/msm/msm.cu.o
+    COMMAND ${CMAKE_OBJCOPY} ARGS --redefine-sym msm_cuda=${CURVE}_msm_cuda ${PROJECT_BINARY_DIR}/CMakeFiles/icicle.dir/appUtils/msm/msm.cu.o
+    COMMAND ${CMAKE_OBJCOPY} ARGS --prefix-symbols=${CURVE}_ ${PROJECT_BINARY_DIR}/CMakeFiles/icicle.dir/appUtils/ntt/ntt.cu.o
```

Comment:
btw - keeping o files in same dir as sources isn't cool cause Rust checks if the dir was changed and rebuilds. We can fine-tune the building later on, but maybe better to change to some dedicated temporary folder right away

Reply:
also `--prefix-symbols=` probably won't work at some point - as it prefixes all symbols - it also brakes some external imports from dynamic libraries - so it fails linking. And for sure it brakes global consts etc. As I tried - so it didn't work on linking the full library - hmm maybe it works ok on individual .cu files until global exports get into library. Instead, objcopy has a file argument and a command to modify only functions listed in that file. We will have to parse all our headers and add `extern` methods to that file. Of course we can add entries there manually - but probably manual editing is not a great idea overall.